app/src/main/java/com/iti/android/team1/ebuy/ui/product_details_screen/viewmodel/AddToCartViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package com.iti.android.team1.ebuy.ui.product_details_screen.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.iti.android.team1.ebuy.model.datasource.repository.IRepository
+import com.iti.android.team1.ebuy.model.pojo.Product
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class AddToCartViewModel(private val repository: IRepository) : ViewModel() {
+
+    private var _plusButtonState: MutableStateFlow<Boolean> = MutableStateFlow(true)
+    val plusButtonsState = _plusButtonState.asStateFlow()
+
+    private var _minusButtonState: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val minusButtonState = _minusButtonState.asStateFlow()
+
+    private var _quantityText: MutableStateFlow<Int> = MutableStateFlow(1)
+    val quantityText = _quantityText.asStateFlow()
+
+    private var quantity = 1
+    private var productQuantity = 0
+
+    fun setProductQuantity(productQuantity: Int) {
+        this.productQuantity = productQuantity
+    }
+
+    fun insertProductToCart(product: Product) {
+        viewModelScope.launch(Dispatchers.IO) {
+            repository.addProductToCart(product, quantity)
+        }
+    }
+
+    fun onPressPlusButton() {
```

Comment:
function onPressPlusButton()  is doing exactly the same factuality of onPressMinusButton(), the only difference between them that onPressPlusButton increases the number in database unlike onPressMinusButton.
It would be much better if we added a parameter to the function called update that updates the number in the database if it is possible. by that, we will get rid of this code duplication.

Reply:
Their logic is clearly different; there is no duplication here; onPressPlusButton checks for at least the maximum amount and changes the state of the button if it is present; onPressMinusButton checks for at least one product and changes the state of the button as well. If I make it simply one function in ViewModel, I will construct two private methods to handle if plus or minus, resulting in three methods instead of two.

Reply:
Its update query updates the date at the end so technically it is the same thing with different conditions, 
you can update your condition to look like this 
if (quantity > 1 || quantity < productQuantity)
emitQuantityState()